src/models/CompanyApplication.js

```diff
@@ -64,18 +79,49 @@ CompanyApplicationSchema.virtual("state").get(function() {
     else return ApplicationStatus.REJECTED;
 });
 
+async function validateEmailUniqueAccount(value) {
+    try {
+        await checkDuplicatedEmail(value);
+    } catch (e) {
+        return false;
+    }
+    return true;
+}
+
+
 function validateDecisionDate(value) {
     return !value || (value > this.submittedAt);
 }
 
-function validateApprovedDate(value) {
+function validateMutuallyExclusiveEvents(field) {
 
-    return !value || !this.rejectedAt;
+    return function(value) {
+        return !value || !this[field];
+    };
 }
 
-function validateRejectedDate(value) {
-    return !value || !this.approvedAt;
+const applicationUniqueness = async (email) => {
+    const existingApplications = await CompanyApplication.find({ email });
+    if (existingApplications.some((application) =>
+        application.state === ApplicationStatus.PENDING ||
+    application.state === ApplicationStatus.APPROVED)
+    ) {
+        throw new Error(CompanyApplicationRules.ONLY_ONE_APPLICATION_ACTIVE_PER_EMAIL.msg);
```

Comment:
In here you could simply return false and mongoose would use the correct message